src/java/org/apache/cassandra/locator/LocalStrategy.java

```diff
@@ -56,16 +59,36 @@ public ReplicationFactor getReplicationFactor()
         return RF;
     }
 
+    private EntireRange getRange(IPartitioner partitioner)
+    {
+        // No need to synchronize here. In the unlikely event of a race, it's
+        // safe and cheap to create duplicates and overwrite in the cache.
+        EntireRange range = perPartitionerRanges.get(partitioner);
```

Comment:
I did it this way as `ConcurrentHashMap::computeIfAbsent` can have a small penalty in blocking other, non contending updates, but since switching to `IdentityHashMap` that's not a concern.